### PR TITLE
Implement full withdraw list

### PR DESCRIPTION
**Pull request by liviavinci**
screen shot 2018-03-26 at 10 26 31 pm
screen shot 2018-03-26 at 10 26 38 pm

### CODE REVIEW

File: src/redux/Topic/saga.js [hunk 1]
```diff
@@ -87,28 +87,37 @@ export function* calculateWinningsHandler() {
...
-      const result = yield call(request, Routes.api.winnings, options);
-      let value;
-      if (result) {
-        const botWon = satoshiToDecimal(result['0']);
-        const qtumWon = satoshiToDecimal(result['1']);
-        value = { botWon, qtumWon };
-      } else {
-        value = { botWon: 0, qtumWon: 0 };
+        // return only winning addresses
+        if (botWon || qtumWon) {
```
Comment (dwalintukan) [inline src/redux/Topic/saga.js]: is this true if botWon === 0 or qtumWon === 0?

File: src/redux/Topic/reducer.js [hunk 2]
```diff
@@ -1,11 +1,13 @@
 import { Map } from 'immutable';
+import _ from 'lodash';
 import actions from './actions';
 
 const initState = new Map({
   betBalances: [],
   voteBalances: [],
   botWinnings: 0,
   qtumWinnings: 0,
+  walletAddressesWithWinnings: [],
```
Comment (dwalintukan) [inline src/redux/Topic/reducer.js]: nit: simpler name, `winningAddresses`

File: src/scenes/Event/scenes/topic.js [hunk 3]
```diff
@@ -1,4 +1,4 @@
-/* eslint react/no-array-index-key: 0 */ // Disable "Do not use Array index in keys" for options since they dont have unique identifier
+/* eslint react/no-array-index-key: 0, no-nested-ternary: 0 */ // Disable "Do not use Array index in keys" for options since they dont have unique identifier
```
Comment (dwalintukan) [inline src/scenes/Event/scenes/topic.js]: nit: is disabling this rule necessary? or more for convenience? suggest we try to follow the preset rules as much as possible.

File: src/scenes/Event/scenes/topic.js [hunk 4]
```diff
@@ -370,23 +422,36 @@ export default class TopicPage extends React.Component {
...
           id="withdrawDetail.alreadyWithdrawn"
           defaultMessage="You have already withdrawn with this address."
         />,
-        warningTypeClass: EventWarningType.Info,
+        warningTypeClass: classes.withdrawn,
+      };
+    }
+
+    // Can withdraw winning
+    const winniningAddresses = _.filter(walletAddressesWithWinnings, {
```
Comment (dwalintukan) [inline src/scenes/Event/scenes/topic.js]: can use `_.find()` for better efficiency. filter will iterate through the entire list (o(n) time), while find will stop at the first element found.

File: src/redux/Topic/saga.js (hunk 1 above)
Comment (liviavinci) [reply src/redux/Topic/saga.js]: do you mean if botWon === 0 and qtumWon === 0? if botWon === 0 and qtumWon === 0 it will return false.

Comment (dwalintukan) [reply src/redux/Topic/saga.js]: k. just wanted to clarify that it checks botWon > 0 || qtumWon > 0 👍

**Pull request merged**